palette/src/encoding/rec_standards.rs

```diff
@@ -0,0 +1,296 @@
+//! The ITU-R Recommendation BT.2020 (Rec. 2020) and BT.709 (Rec. 709) standards and their
+//! associated transfer function.
+
+use crate::{
+    bool_mask::LazySelect,
+    encoding::{FromLinear, IntoLinear, Srgb},
+    luma::LumaStandard,
+    num::{Arithmetics, MulAdd, MulSub, PartialCmp, Powf, Real},
+    rgb::{Primaries, RgbSpace, RgbStandard},
+    white_point::{Any, D65},
+    Mat3, Yxy,
+};
+
+use lookup_tables::*;
+
+mod lookup_tables;
+
+/// The Rec. 2020 standard, color space, and transfer function.
+///
+/// # As transfer function
+///
+/// `Rec2020` will not use any kind of approximation when converting from `T` to
+/// `T`. This involves calls to `powf`, which may make it too slow for certain
+/// applications.
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub struct Rec2020;
+
+impl<T: Real> Primaries<T> for Rec2020 {
+    // Primary values taken from ITU specification:
+    // https://www.itu.int/dms_pubrec/itu-r/rec/bt/R-REC-BT.2020-2-201510-I!!PDF-E.pdf
+    fn red() -> Yxy<Any, T> {
+        Yxy::new(T::from_f64(0.708), T::from_f64(0.292), T::from_f64(0.2627))
+    }
+    fn green() -> Yxy<Any, T> {
+        Yxy::new(T::from_f64(0.170), T::from_f64(0.797), T::from_f64(0.6780))
+    }
+    fn blue() -> Yxy<Any, T> {
+        Yxy::new(T::from_f64(0.131), T::from_f64(0.046), T::from_f64(0.0593))
+    }
+}
+
+impl RgbSpace for Rec2020 {
+    type Primaries = Rec2020;
+    type WhitePoint = D65;
+
+    #[rustfmt::skip]
+    #[inline(always)]
+    fn rgb_to_xyz_matrix() -> Option<Mat3<f64>> {
+        // Matrix calculated using specified primary values and white point
+        // using formulas from http://www.brucelindbloom.com/index.html?Eqn_RGB_XYZ_Matrix.html
+        Some([
+            0.6370102, 0.1446150, 0.1688448,
+            0.2627217, 0.6779893, 0.0592890,
+            0.0000000, 0.0280723, 1.0607577,
+        ])
+    }
+
+    #[rustfmt::skip]
+    #[inline(always)]
+    fn xyz_to_rgb_matrix() -> Option<Mat3<f64>> {
+        // Matrix calculated using specified primary values and white point
+        // using formulas from http://www.brucelindbloom.com/index.html?Eqn_RGB_XYZ_Matrix.html
+        Some([
+             1.7165107, -0.3556417, -0.2533455,
+            -0.6666930,  1.6165022,  0.0157688,
+             0.0176436, -0.0427798,  0.9423051,
+        ])
+    }
+}
+
+impl RgbStandard for Rec2020 {
+    type Space = Rec2020;
+    type TransferFn = RecOetf;
+}
+
+impl LumaStandard for Rec2020 {
+    type WhitePoint = D65;
+    type TransferFn = RecOetf;
+}
+
+/// The Rec. 709 standard, color space, and transfer function.
+///
+/// # As transfer function
+///
+/// `Rec709` will not use any kind of approximation when converting from `T` to
+/// `T`. This involves calls to `powf`, which may make it too slow for certain
+/// applications.
+pub struct Rec709;
```

Comment:
This and `RecOetf` should derive the same traits as other meta types derive. Just in case they end up in generics with those bounds.